packages/ptokens-swap/src/ptokens-swap.ts

```diff
@@ -16,20 +17,23 @@ export class pTokensSwap {
   private _destinationAssets: DestinationInfo[]
   private _amount: BigNumber
   private _controller: AbortController
+  private _interim: NetworkId
 
   /**
    * Create and initialize a pTokensSwap object. pTokensSwap object shall be created using a pTokensSwapBuilder object.
    * @param sourceAsset - The pTokensAsset that will be the source asset for the swap.
    * @param destinationAssets - The pTokensAsset array that will be destination assets for the swap.
    * @param amount - The amount of source asset that will be swapped.
+   * @param interim - The NetworkId of the InterimChain.
    */
-  constructor(sourceAsset: pTokensAsset, destinationAssets: DestinationInfo[], amount: BigNumber.Value) {
+  constructor(sourceAsset: pTokensAsset, destinationAssets: DestinationInfo[], amount: BigNumber.Value, interim: NetworkId) {
```

Comment:
I would put a default here, i.e. INTERIM_CHAIN_NETWORK_ID = NetworkIds.PolygonMainned.
Define the constant in the constants package.
```suggestion
  constructor(sourceAsset: pTokensAsset, destinationAssets: DestinationInfo[], amount: BigNumber.Value, interimNetworkId =  INTERIM_CHAIN_NETWORK_ID) {
```